### PR TITLE
Adding Modifier Composition API

### PR DESCRIPTION
**Pull request by cnstoll**
I’m adding a modifier composition API to build hero modifier animation strings using method chaining. The idea is to create a compiled method for creating an animation. There are a lot of benefits to this:

- Ease of use. It’s a lot easier to build animation strings using autocomplete. It’s easy to see all of the available modifier options for use in an animation.

- You don’t have to remember or debug syntax. The composition does that for you.

- It’s a great foundation to build on for the future. Using this type of interface you can build modifier validation and other features (including pre-fabricated or reusable animations).

- It’s pretty scalable. All you have to do is add another method when you support more animation modifiers.

- It will also work with HeroID tags to specific views. You can pass views as parameters and use their HeroIDs to construct the appropriate animation string.

Basically, I love the idea of Hero. But the string syntax for constructing modifiers was making it harder for me to use. This keeps the flexible string syntax while adding a composition interface that’s more discoverable, safer, and easier to use.

### CODE REVIEW

**Comment by lkzhao** [comment]
@cnstoll We had this conversation in #11 about replacing the string modifiers API. Your builder pattern is great, but we came to the conclusion that we should continue with a pure swift enum implementation. We welcome proposals like this but please join the discussion next time.
Anyway, thank you for your contributions. I would love to have you onboard even though I cannot merge this PR.
